src/mon/NVMeofGwMap.cc

```diff
@@ -436,7 +436,22 @@ int  NVMeofGwMap::find_failover_candidate(const GW_ID_T &gw_id, const std::strin
      {
          case GW_STANDBY_STATE:
          case GW_IDLE_STATE:
-          // nothing to do
+         case GW_BLOCKED_AGROUP_OWNER:
+         {
+           GW_STATE_T* gw_state = find_gw_map(gw_id, nqn);
+            if(grpid == gw_state->optimized_ana_group_id) {// Try to find GW that temporary owns my group - if found, this GW should pass to standby for  this group
+               auto subsyst_it = find_subsystem_map(nqn);
+               for (auto& itr : *subsyst_it){
```

Comment:
possibly use structured binding with meaningful names for the 'first' & 'second'